editors/code/src/commands.ts

```diff
@@ -260,6 +261,33 @@ export function openCargoToml(ctx: CtxInit): Cmd {
     };
 }
 
+export function revealDependency(ctx: CtxInit): Cmd {
+    return async (editor: RustEditor) => {
+        const rootPath = vscode.workspace.workspaceFolders![0].uri.fsPath;
+        const documentPath = editor.document.uri.fsPath;
+        if (documentPath.startsWith(rootPath)) return;
```

Comment:
I'm not entirely sure why you'd want to do this. There's already non-workspace filtering done on the server; this just adds an additional check.

(This also breaks for monorepos where users open their editor at the monorepo root. Non-workspace crates are often stored in `third-party/` but share the workspace root.)

Reply:
The purpose of the function revealDependency is to make the TreeView "reveal" the opened file in the editor. Example:
![image](https://user-images.githubusercontent.com/5748995/229884702-cc765e2a-3e48-440e-9937-b925f2844bd7.png)

But if I where to open a file that belongs to the local project, I dont want to trigger any "revealing"

Reply:
> The purpose of the function revealDependency is to make the TreeView "reveal" the opened file in the editor. 

Gotcha, thanks for explaining—I've built the branch locally and usage made a bit more sense.

> But if I where to open a file that belongs to the local project, I dont want to trigger any "revealing"

That makes sense and works for most default Cargo-based builds. However, using `if (documentPath.startsWith(rootPath)) return;` in situations where `cargo vendor` has been run _or_ when a user opens their VS Code instance at the root of the monorepo (where a `third-party/` folder exists and contains vendored Rust crates), this check will prevent this tree-based explorer from ever opening and it seems unfortunate that those users won't benefit from this feature[^1]. I don't have merge permissions on rust-analyzer, but I'd recommend removing this check, as `fetch_crates` in `crates/ide/src/fetch_crates.rs` already performs a similar check. I need to look over this PR a bit more, but if a user asks to reveal the dependency for a crate that is part of their editable workspace, then I would consider showing it. Alternatively, it might make sense to add another LSP extension to query the server itself to check whether a file is editable/part of the current editable workspace or not. 

[^1]: Personally, I'd love to build atop of this UI in order to allow users to incrementally add crates to their (editable) rust-analyzer workspace—it's a pretty workflow in monorepos.

Reply:
@davidbarsky thanks for the reply, I just pushed a commit improving some pain points and hopefully giving it proper support to multiple workspace folders.
Could you point me at any github repos that make use of the features that you mentioned? I would love to test this cases because I suspect that they won't be a problem, because if they're already part of the project tree, the main treeView of vscode will already display those dependencies.

Reply:
Just to add to my previous comment, I created a vendored project and everything seems fine.
![image](https://user-images.githubusercontent.com/5748995/230171372-3645d20c-22d8-4c5d-83be-4bb691e2fa40.png)

Reply:
> Could you point me at any github repos that make use of the features that you mentioned?

@bruno-ortiz I can't think of any project off-hand, but it would be possible to take, e.g., the rust-analyzer repository and run `cargo vendor` and you'll see a subset of the behavior I'm mentioning.

> I would love to test this cases because I suspect that they won't be a problem, because if they're already part of the project tree, the main treeView of vscode will already display those dependencies.

At the risk of being *that person*, but a smaller repository won't really demonstrate this issue—it only really becomes apparent once a monorepo has tens (or hundreds!) of millions of lines of code. While said dependencies _are_ technically part of the primary explorer view, it's often infeasible to see them at a glance because of the sheer number of folders present in the default file explorer.

My mental model of this feature—at least in monorepos—is that it is a subset of the monorepo (think of it as a materialized view...) that corresponds to the portions of monorepo most relevant to the code being worked on at the moment.

The reason that I'm so adamant about requesting this is that I didn't really _get_ the need/desire for it until I tried it. While it is technically possible to navigate through the primary explorer window to dependencies, it ends up taking closer to 30-60 seconds per dependency rather than something instantaneous.